test/RouteCollectorTest.php

```diff
@@ -185,13 +186,66 @@ public function testCreatingHttpRouteMethodWithExistingPathButDifferentMethodCre
         $this->assertSame($middleware, $test->getMiddleware());
     }
 
+    public function testCreatingHttpRouteWithExistingPathShouldBeLinear()
+    {
+        $start = microtime(true);
+        foreach (range(1, 10) as $item) {
+            $this->collector->get("/bar$item", $this->noopMiddleware);
+        }
+        $baseDuration = microtime(true) - $start;
+        $this->collector = new RouteCollector($this->router->reveal());
+
+        $start = microtime(true);
+        foreach (range(1, 10000) as $item) {
+            $this->collector->get("/foo$item", $this->noopMiddleware);
+        }
+
+        $duration = microtime(true) - $start;
+        $expectedDuration = $baseDuration * 1000;
+        $error = 30 * $expectedDuration / 100;
+        $this->assertTrue(
+            $expectedDuration + $error > $duration,
+            "Route add time should be linear by amount of routes" .
+            ", expected duration: $expectedDuration" .
+            ", possible error: $error" .
+            ", actual duration: $duration"
```

Comment:
Please use `sprintf()` here to create the failure string. If the string needs to flow to multiple lines, put the concatenation operator on the next line, at the same level of indentation:

```php
"Route add time should be linear by amount of routes,"
. " expected duration: %d,"
. " possible error: %d,"
. " actual duration: %d"
```